lib/zlib.js

```diff
@@ -592,6 +597,10 @@ function processCallback() {
     assert(false, 'have should not go down');
   }
 
+  if (self.destroyed) {
```

Comment:
There is already a check for this a few line above, isn't that one sufficient?

Reply:
I guess the idea is that the `push()` call can trigger `on('data')` listeners which can destroy the stream?

Either way, it would be great to have a test for this (and the other bits in this PR)

Reply:
Ok, makes sense, I think a comment would also help.

Reply:
This is actually already tested as the existing tests fail (crash even!) without this.